strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java

```diff
@@ -16,10 +16,10 @@
 public interface ArtifactManagementService extends ConfigurationService
 {
 
-    void store(String storageId,
-               String repositoryId,
-               String path,
-               InputStream is)
+    void validateAndStore(String storageId,
```

Comment:
Honestly, I don't like it too.

Please compare [master version of AbstractArtifactManagementService.java](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/AbstractArtifactManagementService.java) with [updated in this branch AbstractArtifactManagementService.java](https://github.com/fuss86/strongbox/blob/ff35f4923aa89a593136e9265b0e3f75d7328060/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/AbstractArtifactManagementService.java) in terms of `store` and `validateAndStore` methods. 

In master branch, both `store` methods perform validation (`performRepositoryAcceptanceValidation`). Moreover, first `store` method invokes the second `store`, so the validation is performed twice unnecessarily. In master branch, the second `store` method is additionally invoked while downloading artifact from remote repository. As I explained it to you, validation (`performRepositoryAcceptanceValidation`) is also performed there in this case, over the remote repository, which is wrong. Those 2 cases show that the validation performed in the second `store` method is unnecessary and wrong. Therefore I removed it from this method in my branch.

But still first `store` method performs validation (`performRepositoryAcceptanceValidation`). So these 2 methods cannot be named the same. That's why I choose this name. 

What better solution do you suggest ?